models/perpetuals/optimism/perpetual_v2_optimism.sql

```diff
@@ -0,0 +1,73 @@
+{{ config(
+	schema = 'perpetual_v2_optimism',
+	partition_by = ['block_time'],
+	materialized = 'incremental',
+	file_format = 'delta',
+	incremental_strategy = 'merge',
+	unique_key = ['block_time', 'trade_id']
+	)
+}}
+
+WITH perps AS (
+	SELECT
+		p.evt_block_time AS block_time
+		,p.baseToken
+		,pp.pool AS market_address
+		,ABS(p.exchangedPositionNotional)/1e18 AS volume_usd
+		,p.fee/1e18 AS fee_usd
+		,co.output_0/1e6 AS margin_usd
+
+		,CASE
+		WHEN CAST(p.exchangedPositionSize AS DOUBLE) > 0 THEN 'long'
+		WHEN CAST(p.exchangedPositionSize AS DOUBLE) < 0 THEN 'short'
+		ELSE 'NA'
+		END AS trade
+
+		,'Perpetual' AS project
+		,'2' AS version
+		,p.trader
+		,p.exchangedPositionNotional AS volume_raw
+		,p.evt_tx_hash AS tx_hash
+		,p.evt_index
+	FROM {{ source('perp_v2_optimism', 'ClearingHouse_evt_PositionChanged') }} AS p
+	LEFT JOIN {{ source('perp_v2_optimism', 'Vault_call_getFreeCollateralByRatio') }} AS co
+		ON p.evt_tx_hash = co.call_tx_hash
+	{% if is_incremental() %}
+	WHERE p.evt_block_time >= (SELECT MAX(block_time) FROM {{ this }})
```

Comment:
We've settled on having incremental filters use this convention:
	`WHERE p.evt_block_time >= date_trunc("day", now() - interval '1 week')`
	I'll point to places where this should be replaced